plugin/pkg/scheduler/schedulercache/node_info.go

```diff
@@ -153,6 +165,18 @@ func calculateResource(pod *api.Pod) (cpu int64, mem int64, non0_cpu int64, non0
 	return
 }
 
+// Sets the overall node information.
+func (n *NodeInfo) SetNode(node *api.Node) error {
+	n.node = node
+	return nil
+}
+
+// Removes the overall information about the node.
+func (n *NodeInfo) RemoveNode(node *api.Node) error {
```

Comment:
>  since pods & nodes are delivered in different watch channels, there may be a race and node deletion may be delivered first. 

Ah. I see. That sounds not right though. If a node is deleted, but we still know of some pods on it... Is there any explicit docs or discussion on this?